src/main/java/com/uid2/shared/auth/OperatorKey.java

```diff
@@ -26,14 +27,15 @@ public OperatorKey(String key, String name, String contact, String protocol, lon
         this.siteId = null;
     }
 
-    public OperatorKey(String key, String name, String contact, String protocol, long created, boolean disabled, Integer siteId) {
+    public OperatorKey(String key, String name, String contact, String protocol, long created, boolean disabled, Integer siteId, Set<Role> roles) {
         this.key = key;
         this.name = name;
         this.contact = contact;
         this.protocol = protocol;
         this.created = created;
         this.disabled = disabled;
         this.siteId = siteId;
+        this.roles = Collections.unmodifiableSet(roles);
```

Comment:
are you going to require user to explicitly specify Role.OPERATOR in `roles` param (in admin service)? or is it better to hardcode the Role.OPERATOR here as Operator must have Role.OPERATOR anyway?

Reply:
for backward compatibility reason again, might want to add another constructor instead of just adding the `roles` param to this constructor. again might want to hardcode Role.OPERATOR in this roles-less constructor call

Reply:
Currently, in admin service I was using the first constructor which not required to set siteId and roles, and chain it with withRoles if we want to add a role to the optout key. So I am not sure if we need another constructor? But I totally agree we should make Role.OPERATOR to be the default role

Reply:
i meant when you make the respective change in admin service (which will allow setting additional OPTOUT role), it should call such "another constructor" instead of extending the existing one like above, for backward compatibility.

Reply:
Oh, I see what you mean, I will add the new constructor